src/metabase/api/dashboard.clj

```diff
@@ -88,26 +107,36 @@
                    (not= embedding_params (:embedding_params dash))))
       (check-embedding-enabled)
       (check-superuser)))
-  (check-500 (-> (assoc dashboard :id id)
-                 (dashboard/update-dashboard! *current-user-id*))))
-
-
+  (check-500
+   (db/update! Dashboard id
+     ;; description, position are allowed to be `nil`. Everything else must be non-nil
+     (u/select-keys-when dashboard
+       :present #{:description :position}
+       :non-nil #{:name :parameters :caveats :points_of_interest :show_in_getting_started :enable_embedding :embedding_params :archived})))
+  ;; now publish an event and return the updated Dashboard
+  (u/prog1 (Dashboard id)
+    (events/publish-event! :dashboard-update (assoc <> :actor_id *current-user-id*))))
+
+
+;; TODO - We can probably remove this in the near future since it should no longer be needed now that we're going to be setting `:archived` to `true` via the `PUT` endpoint instead
 (defendpoint DELETE "/:id"
   "Delete a `Dashboard`."
   [id]
+  (log/warn "DELETE /api/dashboard/:id is deprecated. Instead of deleting a Dashboard, you should change its `archived` value via PUT /api/dashboard/:id.")
   (let [dashboard (write-check Dashboard id)]
-    (db/delete! Dashboard :id id)
+    (db/delete! Dashboard id)
     (events/publish-event! :dashboard-delete (assoc dashboard :actor_id *current-user-id*)))
   generic-204-no-content)
 
 
+;; TODO - param should be `card_id`, not `cardId` (fix here + on frontend at the same time)
```

Comment:
that's a TODO for some other time, no need to tackle it here in this PR

as I'm working on things I add notes like that to go back and clean up later but tackling them would be out of scope for this PR